src/components/PupilViews/PupilIntro/PupilIntro.view.tsx

```diff
@@ -46,12 +48,16 @@ export const PupilViewsIntro = (props: PupilViewsIntroProps) => {
     currentSection,
     updateSectionResult,
     sectionResults,
+    proceedToNextSection,
+    lessonReviewSections,
   } = useLessonEngineContext();
   const getSectionLinkProps = useGetSectionLinkProps();
+  const { trackSectionStarted } = useTrackSectionStarted();
   const { startDownload, isDownloading } = useWorksheetDownload(
     lessonSlug,
     isLegacy ?? false,
   );
+  const { track } = usePupilAnalytics();
 
   useEffect(() => {
```

Comment:
BTW: this is another example of some code that doesn't need to be inside useEffect

Reply:
i did think about that when i saw it while doing this ticket